pyodide_build/common.py

```diff
@@ -16,12 +16,14 @@
 from collections import deque
 from collections.abc import Generator, Iterable, Iterator, Mapping, Sequence
 from contextlib import contextmanager
+from functools import lru_cache
```

Comment:
```suggestion
from functools import cache
```

From @hoodmane's suggestion on using `@functools.cache()` above.